app/routes/participate/survey.js

```diff
@@ -39,8 +39,10 @@ export default Ember.Route.extend(AuthenticatedRouteMixin, {
           }
           this.set('_experiment', experiment);
           session.set('experimentVersion', '');
-          session.set('locale', this.controllerFor('participate').get('locale'));
-          session.set('studyId', this.controllerFor('participate').get('studyId'));
+          session.set('extra', {
+              locale: this.controllerFor('participate').get('locale'), // The user's locale
```

Comment:
Just to avoid refactorings and prefer one intuitive way of doing things: can we instead get the locale from the service, `i18n.locale`?

The data flow is a bit convoluted but here's where it eventually gets set:
https://github.com/abought/isp/blob/cc44c7a42facb03c0894442d057658fc3496fc40/app/components/language-picker/component.js#L37-L37

As for studyId- you could get this from the participant record directly(`account.extra....`), but in this case there isn't one unambiguous source of truth, so this is as good as any.